### ISSUE TITLE
[Epic] De-couple STORAGE and COMPUTE

### ISSUE THREAD

**Issue by frankmcsherry** [MaterializeInc/materialize#11434]
### Feature request

**updated on 2022.04.11 by @nmeagan11**

It is a main requirement of Platform to separate the storage layer from the compute layer, which gives compute instances independent access to sources without any duplication of the sources and without interference from other compute instances.

Beyond the above, the STORAGE layer currently presents to COMPUTE through a single server that must explicitly transit all of the data. This is a scalability bottleneck, and general performance liability. We should correct it to rely instead on a scalable storage medium.

**Comment by petrosagg** [comment]
* [ ] https://github.com/MaterializeInc/materialize/issues/11905
* [ ] https://github.com/MaterializeInc/materialize/issues/11906

**Comment by nmeagan11** [comment]
PR: https://github.com/MaterializeInc/materialize/pull/12216

**Comment by benesch** [comment]
#12216 is feature complete, but blocked on some timeouts in CI. We're hopeful that #12425 will fix those timeouts, but we won't know for sure until #12216 gets rebased on top of #12425. If we're lucky, though, perhaps as soon as tomorrow!

**Comment by benesch** [comment]
This is done! There are some known performance bottlenecks (only one persist shard per source), but we can track those in a new issue for M2.
